src/test/java/com/snowflake/kafka/connector/SnowflakeSinkTaskStreamingDLQTest.java

```diff
@@ -0,0 +1,100 @@
+package com.snowflake.kafka.connector;
+
+import com.snowflake.kafka.connector.builder.SinkRecordBuilder;
+import com.snowflake.kafka.connector.dlq.InMemoryKafkaRecordErrorReporter;
+import com.snowflake.kafka.connector.internal.TestUtils;
+import com.snowflake.kafka.connector.records.SnowflakeJsonConverter;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import net.snowflake.ingest.streaming.InsertValidationResponse;
+import net.snowflake.ingest.streaming.OpenChannelRequest;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestChannel;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+
+public class SnowflakeSinkTaskStreamingDLQTest {
```

Comment:
i would add one more test case:
when we throw exception that has different error code and is not sent to DLQ but rethrown

Reply:
There is no obvious code path to provide such test at task level (checked the one with null records but it turned out to be dead code).
It is possible to mock RecordService to achieve such behaviour but it requires a lot of additional refactoring. I think it is an overkill for that particular fix.